src/operation.rs

```diff
@@ -473,6 +491,22 @@ impl Operation {
                     .send(Message::PendingProgress(id, 100.0))
                     .await;
             }
+            Self::Extract { paths, to } => {
+                for path in paths {
+                    let to = path.file_stem().map(|stem| to.join(stem)).unwrap();
+                    tokio::task::spawn_blocking(move || {
+                        process::Command::new("unzip")
```

Comment:
An implementation of this should not call out to an external program

Reply:
Thank you for looking at this. Do you mean I should import a library to handle extracting files instead of using unzip? Are there any in particular you'd trust to use for this?

Reply:
I'm updating it to use the `zip` crate, which looks good for this.

Reply:
That is fine. More archive types should be supported if possible.

Reply:
Definitely want to see more types, and I might keep going on this, but I wanted to keep the scope here small but usable at first. 

Let me know how this looks. I changed it to use the `zip` crate, and also changed it to create duplicate output folders if you extract multiple times, like the copy function.